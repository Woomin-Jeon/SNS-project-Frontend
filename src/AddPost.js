/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable no-alert */
/* eslint-disable no-shadow */
import React, { useState } from 'react';
import { addPost } from './function';
import ShowPost from './ShowPost';

const initialPost = {
  temptId: '',
  temptContents: '',
  post: [
    {
      id: '홍길동', // 작성자
      contents: '오늘은 날씨가 쌀쌀하니 좋아요 기능을 넣어보면 좋을 것 같다', // 게시글의 내용
      thumbCount: 0, // 좋아요 갯수
      sharingCount: 0, // 공유 횟수
      commentCount: 0, // 댓글 갯수
      comment: '',
    },
  ],
};
/* 댓글 입력 버튼을 누르면, 해당 게시글의 id값을 onclick함수의 매개변수로 넣어주고 id를 포함한 객체를 comment
배열에 넣어준다. 그리고 post.map에서 post.id == comment.id 인 것을 찾아서 이것의 statement를 렌더링 해준다.
*/
function AddPost() {
  const [poststate, setPostState] = useState(initialPost);
  const { temptId, temptContents } = poststate;

  const setPostTemptId = (temptId) => {
    setPostState({ ...poststate, temptId });
  };
  const setPostTemptContents = (temptContents) => {
    setPostState({ ...poststate, temptContents });
  };

  const handleAddPost = () => {
    if (temptId.trim() && temptContents.trim()) {
      setPostState({ ...addPost(poststate, temptId, temptContents), temptId: '', temptContents: '' });
    } else {
      alert('내용을 입력해주세요');
    }
  };

  return (
    <>
      <div className="addpost">
        <div className="addpost-title">게시물 만들기</div>
        <div>작성자: <input className="addpost-writer" type="text" value={temptId} onChange={(e) => setPostTemptId(e.target.value)} /></div>
        <div>내용: <input className="addpost-inputcontents" type="text" value={temptContents} onChange={(e) => setPostTemptContents(e.target.value)} /></div>
        <button className="addpost-out" type="button" onClick={handleAddPost}>게시</button>
      </div>
      <ShowPost poststate={poststate} />
    </>
  );
}

export default AddPost;
