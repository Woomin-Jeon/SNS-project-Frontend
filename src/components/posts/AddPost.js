/* eslint-disable no-shadow */
/* eslint-disable react/prop-types */
/* eslint-disable no-alert */
import React, { useState } from 'react';
import { addPost } from '../../apis/service';

function AddPost({
  currentUserState,
  postState,
  setPostState,
}) {
  const [temptState, setTemptState] = useState('');
  const { userName, id } = currentUserState;

  const setPostTemptContents = (temptState) => {
    setTemptState(temptState);
  };

  const handleAddPost = async () => {
    const { timeLinePosts } = await addPost(id, userName, temptState);
    const { post } = timeLinePosts;
    const apiPost = post;

    console.log(apiPost);
    if (temptState.trim()) {
      setPostState({ ...postState, post: [...apiPost]})
      setTemptState('');
    } else {
      alert('내용을 입력해주세요');
    }
  };

  return (
    <div className="addpost">
      <div className="addpost-title">게시물 만들기</div>
      <div className="addpost-notion">
        <span className="addpost-span">
          {userName}
        </span>
        님, 무슨 생각을 하고 계신가요?
      </div>
      <div className="addpost-notion">생각을 게시글로 표현 해주세요</div>
      <input className="addpost-inputcontents" type="text" value={temptState} onChange={(e) => setPostTemptContents(e.target.value)} />
      <button className="addpost-out" type="button" onClick={handleAddPost}>게시</button>
    </div>
  );
}

export default AddPost;
