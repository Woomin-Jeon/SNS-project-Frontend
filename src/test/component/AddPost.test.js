import React from 'react';
import {shallow} from "enzyme";
import AddPost from "../../components/posts/AddPost";

describe('AddPost', () => {
  let currentUserState;
  let postState;
  let setPostState;

  beforeEach(() => {
    currentUserState = { userName: 'Woomin' };
  });

  it('renders well', () => {
    const component = shallow(
      <AddPost
        currentUserState={currentUserState}
        postState={postState}
        setPostState={setPostState}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.text()).toMatch('Woomin');
    expect(component.find('button').text()).toBe('게시');
    expect(component.find('form').text()).toBeDefined();
  });
});
