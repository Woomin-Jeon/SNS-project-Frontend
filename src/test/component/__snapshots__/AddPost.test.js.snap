// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPost renders well 1`] = `ShallowWrapper {}`;
